src/types/infer.ts

```diff
@@ -2662,4 +2663,90 @@ export class InferType {
 
         return resolvedCalleeT;
     }
+
+    private isExternalCallContext(expr: Expression): boolean {
+        if (expr instanceof Identifier) {
+            const exprT = this.typeOf(expr);
+
+            if (exprT instanceof UserDefinedType) {
+                if (exprT.definition instanceof ContractDefinition) {
+                    return true;
+                }
+            }
+
+            if (exprT instanceof TypeNameType) {
+                return (
+                    exprT.type instanceof UserDefinedType &&
```

Comment:
I am still confused about this one. I thought there may be cases when `Library.fun(....)` may be internal? Or is it always external if the function is defined as public? Sorry I may be forgetting  here

Reply:
Addressed in 4ea38d2. We consider treating such call as `external`.